File: frontend-pantry-android/app/src/main/java/com/zancheema/android/pantry/ui/login/LoginService.java
package com.zancheema.android.pantry.ui.login;

import com.zancheema.android.pantry.ui.login.dto.AuthToken;

import retrofit2.Call;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface LoginService {
    @POST("auth/token")
    Call<AuthToken> getToken(@Header("Authorization") String authHeader);
}

File: frontend-pantry-android/app/src/main/java/com/zancheema/android/pantry/ui/login/LoginService.kt
package com.zancheema.android.pantry.ui.login

import com.zancheema.android.pantry.ui.login.dto.AuthToken
import retrofit2.Response
import retrofit2.http.Header
import retrofit2.http.POST

interface LoginService {
    @POST("auth/token")
    suspend fun getToken(@Header("Authorization") authHeader: String?): Response<AuthToken>
}